Mathlib/Tactic/Linter/Multigoal.lean

```diff
@@ -0,0 +1,161 @@
+/-
+Copyright (c) 2024 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+import Lean.Elab.Command
+
+/-!
+#  The "multiGoal" linter
+
+The "multiGoal" linter emits a warning where there is more than a single goal in scope.
+There is an exception: a tactic that closes *all* remaining goals is allowed.
+
+There are a few tactics that are intended to work specifically in such a situation and the linter
+ignores them. Otherwise, whenever a tactic leaves multiple goals, the linter will emit a warning,
+unless some form of "focusing" tactic is used.
+Typically, the focusing is achieved by the `cdot`: `·`, but, e.g., `focus` or `on_goal x` also
+serve a similar purpose.
+
+TODO:
+* Should the linter flag unnecessary scoping as well?
+  For instance, should
+  ```lean
+  example : True := by
+    · · exact .intro
+  ```
+  raise a warning?
+* Custom support for "accumulating side-goals", so that once they are all in scope
+  they can be solved in bulk via `all_goals` or a similar tactic.
+* In development, `on_goal` has been partly used as a way of silencing the linter
+  precisely to allow the accumulation referred to in the previous item.
+  Maybe revisit usages of `on_goal` and also nested `induction` and `cases`.
+-/
+
+open Lean Elab
+
+namespace Mathlib.Linter
+
+/-- The "multiGoal" linter emits a warning when there are multiple active goals. -/
+register_option linter.style.multiGoal : Bool := {
+  defValue := false
+  descr := "enable the multiGoal linter"
+}
+
+namespace Style.multiGoal
+
+/-- The `SyntaxNodeKind`s in `exclusions` correspond to tactics that the linter allows,
+even though there are multiple active goals.
+Reasons for admitting a kind in `exclusions` include
+* the tactic focuses on one goal, e.g. `·`, `focus`, `on_goal i =>`, ...;
+* the tactic is reordering the goals, e.g. `swap`, `rotate_left`, ...;
+* the tactic is structuring a proof, e.g. `skip`, `<;>`, ...;
+* the tactic is creating new goals, e.g. `constructor`, `cases`, `induction`, ....
+
+Tactic combinators like `repeat` or `try` are a mix of both.
```

Comment:
Should this sentence move after the "There is some overlap in scope between `ignoreBranch` and `exclusions`."? (Otherwise, what does "of both" refer to?)